FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m

```diff
@@ -1396,7 +1396,9 @@ - (void)testSetFetchTimeoutConfigSetting {
 - (void)testConfigureConfigWithValidInput {
   // Configure the default app with our options and ensure the Remote Config instance is set up
   // properly.
-  XCTAssertNoThrow([FIRApp configureWithOptions:[self firstAppOptions]]);
+  if (![FIRApp isDefaultAppConfigured]) {
```

Comment:
Why the default app may be configured here? It looks cleaner to me to make sure previous tests clean up the state in `tearDown` and assert this condition in the tests or `setUp`, so the check below is performed unconditionally.

Reply:
The IID SDK's init needs a default app configured. How do i reset FIRApp state in teardown?

Reply:
I think [[FIRApp deleteApp:]](https://github.com/firebase/firebase-ios-sdk/blob/7108b9b08a551d2b96d57800b385b0fd24ebc31d/FirebaseCore/Sources/Public/FIRApp.h#L98) or probably better  [[FIRApp resetApps]](https://github.com/firebase/firebase-ios-sdk/blob/7108b9b08a551d2b96d57800b385b0fd24ebc31d/FirebaseCore/Sources/Private/FIRAppInternal.h#L153) should do the trick.

Reply:
Nice! done.